futures-await-await-macro/src/lib.rs

```diff
@@ -13,16 +13,26 @@ macro_rules! await {
     ($e:expr) => ({
         let mut future = $e;
         loop {
-            match ::futures::Future::poll(&mut future) {
-                ::futures::__rt::std::result::Result::Ok(::futures::Async::Ready(e)) => {
+            let poll = unsafe {
+                let pin = ::futures::__rt::anchor_experiment::Pin::new_unchecked(&mut future);
```

Comment:
The plan is to replace this with the stack-pinning API eventually, right?

Reply:
Maybe we should just do that now.

Reply:
+1